### PR TITLE
Allocate node IP from correct range after the node itself was already allocated

### PR DESCRIPTION
**Pull request by jsuchome**
partially backporting https://github.com/crowbar/crowbar-core/pull/78
Second part is https://github.com/crowbar/barclamp-provisioner/pull/459

Note that now I didn't adapt network_service to keep the changes small(er).

### CODE REVIEW

**Comment by aplanas** [comment]
+1 tested in C5

File: chef/cookbooks/barclamp/libraries/barclamp_library.rb [hunk 1]
```diff
@@ -35,6 +35,7 @@ def self.get_network_by_type(node, type)
               return Network.new(net, data, intf, interface_list)
             end
           end
+          return nil
```
Comment (aplanas) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb:38]: Do we need that?

Comment (jsuchome) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb:38]: Yes, for this
https://github.com/crowbar/barclamp-provisioner/pull/459/files#diff-531ca46b175571ed356cbc6057f38685R139

Comment (tboerger) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb:38]: Hum, no... It's useless as nil will be returned anyway.

Comment (jsuchome) [inline chef/cookbooks/barclamp/libraries/barclamp_library.rb:38]: It is not, at least it wasn't when I played with that.
And current (master) code has it as well, see https://github.com/crowbar/crowbar-core/commit/0fa9984c9355c79f652d19a993c039ee27b90291

**Comment by vuntz** [comment]
@jsuchome do we want to use the same changes that went into master instead of this?

**Comment by jsuchome** [comment]
I don't know. I intentionally made this simpler, so we do not introduce new migration into Cloud5. That move to network transition makes sense, but is not necessary for the fix.

But if we want to really have the same behavior, why not?

**Comment by vuntz** [comment]
> I intentionally made this simpler, so we do not introduce new migration into Cloud5.

Makes sense.

**Comment by vuntz** [comment]
+1

**Comment by jsuchome** [comment]
rebased

**Comment by rhafer** [comment]
@jsuchome Please add a bsc# reference to the commit message's summary.

**Comment by rhafer** [comment]
+1, once the bsc reference is there.

**Pull request merged**